src/com/google/ProductionInstrumentationReporter.java

```diff
@@ -0,0 +1,464 @@
+     * the predicate. As an example, it can return a list of all unique parameters which match a
+     * specific file name.
+     */
+    private List<String> getAllMatchingValues(Predicate<String> comparisonPredicate) {
+      List<String> result = new ArrayList<>();
+      for (String key : parameterMapping.keySet()) {
+        if (comparisonPredicate.test(key)) {
+          result.add(key);
+        }
+      }
+      return result;
+    }
+
+  }
+
+  /**
+   * The instrumentation will send the report in a JSON format where the JSON is a dictionary of the
+   * encoded params to an object of the data collected. Initially this will just be the frequency.
+   */
+  private static class ProfilingData {
+
+    public int frequency;
+
+    public ProfilingData(int frequency) {
+      this.frequency = frequency;
+    }
+
+  }
+
+  /**
+   * This class contains a detailed breakdown of each instrumentation point. For any encoded param,
+   * this class will contain its details. The `executed` property will be a percent of total times
+   * the instrumentation point was called across all provided instrumentation reports. i.e if
+   * this.executed = 100, then every report would have this point present. Similarly, ProfilingData
+   * takes the average of all ProfilingData's for this instrumentation point in all reports.
+   */
+  private static class ProfilingResult {
+
+    public String param;
+    public InstrumentationType type;
+    public int lineNo;
+    public int colNo;
+    public float executed;
+    public ProfilingData data;
+
+    ProfilingResult(InstrumentationMapping instrumentationMapping, String param) {
+      this.param = param;
+      this.type = instrumentationMapping.getType(param);
+      this.lineNo = instrumentationMapping.getLineNo(param);
+      this.colNo = instrumentationMapping.getColNo(param);
+    }
+
+  }
+
+  /**
+   * A class which groups profiling results by source file and contains a mapping of them to each
+   * function which is present in that source file.
+   */
+  private static class ProfilingResultByFile {
+
+    public String fileName;
+    Map<String, List<ProfilingResult>> profilingDataPerFunction;
+  }
+
+  /**
+   * A class that maintains aggregate information which will be converted to a JSON.
+   */
+  private static class AggregateProfilingResults {
+
+    int totalReportsParsed;
+    float percentOfFunctionsExecuted;
+    float percentOfBranchesExecuted;
+
+    List<ProfilingResultByFile> result;
+
+    public AggregateProfilingResults(int totalReportsParsed, List<ProfilingResultByFile> result) {
+      this.totalReportsParsed = totalReportsParsed;
+      this.result = result;
+
+      getPercentOfInstrumentationExecuted();
+    }
+
+    /**
+     * Iterates over the result property and calculates what percent out of all functions and
+     * branches where executed to get aggregated instrumentation data.
+     */
+    private void getPercentOfInstrumentationExecuted() {
```

Comment:
Function is called `get` while it doesn't return anything. It should be called `calculate` or something like that instead.